modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTopologyFutureAdapter.java

```diff
@@ -92,11 +97,14 @@ protected final CacheGroupValidation validateCacheGroup(CacheGroupContext grp, C
 
         CacheGroupContext grp = cctx.group();
 
-        if (cctx.shared().readOnlyMode() && !read && !isSystemCache(cctx.name())
-            && !VOLATILE_DATA_REGION_NAME.equals(cctx.group().dataRegion().config().getName())) {
-            return new CacheInvalidStateException(
-                new IgniteClusterReadOnlyException("Failed to perform cache operation (cluster is in " +
-                    "read-only mode) [cacheGrp=" + cctx.group().name() + ", cache=" + cctx.name() + ']'));
+        if (cctx.shared().readOnlyMode() && !read && !isSystemCache(cctx.name())) {
+            String regionName = grp.dataRegion() == null ? null : grp.dataRegion().config().getName();
+
+            if (!VOLATILE_DATA_REGION_NAME.equals(regionName)) {
```

Comment:
from ticket documentation : _Datastructures can't be used in read-only mode._ why we allow VOLATILE_DATA_REGION_NAME it here ?

Reply:
Good catch! Fixed. The tests are already written in GG-29471